sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -1026,27 +1067,37 @@ void gnrc_rpl_send_DAO(gnrc_rpl_instance_t *inst, ipv6_addr_t *destination, uint
         return;
     }
     me = &netif->ipv6.addrs[idx];
-
-    /* add external and RPL FT entries */
-    /* TODO: nib: dropped support for external transit options for now */
-    void *ft_state = NULL;
-    gnrc_ipv6_nib_ft_t fte;
-    while (gnrc_ipv6_nib_ft_iter(NULL, dodag->iface, &ft_state, &fte)) {
-        DEBUG("RPL: Send DAO - building transit option\n");
-
-        if ((pkt = _dao_transit_build(pkt, lifetime, false)) == NULL) {
+    if (mop == GNRC_RPL_MOP_NON_STORING_MODE) {
+        /* Add the prefix to include the parent's global IP address. */
+        memcpy(parent, &dodag->dodag_id, 8);
```

Comment:
Why do we want to overwrite our parent's address with the root node's address?

Reply:
Okay, I am not super familiar with this part but sounds reasonable.

For modifying the prefix you can use `ipv6_addr_init_prefix` instead of doing the manual copy.

Also, you currently directly overwrite the parent's address stored in the `dodag` structure (`parent` is a reference `&dodag->parent->addr`). You probably want to copy the address before modifying it, otherwise it can mess up other logic.